### PR TITLE
Gtests

### PR DESCRIPTION
**Pull request by velo**
last numbers:
````Tests run: 818, Failures: 2, Errors: 12, Skipped: 369````

now we have pretty much ALL tests on OrientGraphStructureStandardIT fixed!

the two failures are true bugs on this driver

### CODE REVIEW

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java [hunk 1]
```diff
@@ -252,19 +267,34 @@ private OIndexManager getIndexManager() {
...
-    protected static ORecordId createRecordId(Object id) {
+    private ORID checkId(ORID id) {
+        if (!id.isValid())
+            throw new IllegalArgumentException("Invalid id " + id);
+        try {
+            database.getRecordMetadata(id);
+        } catch (IllegalArgumentException e) {
+            // bummer, the API force me to break the chain =((
+            // https://github.com/apache/incubator-tinkerpop/commit/34ec9e7f60f15b5dbfa684a8e96668d9bbcb6752#commitcomment-14235497
+            throw Graph.Exceptions.elementNotFound(Edge.class, id);
```
Comment (velo) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java]: FWIW
https://issues.apache.org/jira/browse/TINKERPOP3-945

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java [hunk 2]
```diff
@@ -238,6 +249,10 @@ private OIndexManager getIndexManager() {
 
     @Override
     public Iterator<Edge> edges(Object... edgeIds) {
+        Set<Class<?>> edgeIdTypes = Stream.of(edgeIds).map(edgeId -> edgeId.getClass()).collect(toSet());
+        if(edgeIdTypes.size() > 1 && edgeIdTypes.contains(OrientEdge.class))
+            throw Graph.Exceptions.idArgsMustBeEitherIdOrElement();
```
Comment (velo) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java]: @mpollmeier 
whould like to hears your thoughts on this....

the structure tests expect this validation

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientElement.java [hunk 3]
```diff
@@ -55,6 +56,8 @@ public Graph graph() {
     }
 
     public <V> Property<V> property(final String key, final V value) {
+        if(value == null)
+            throw Property.Exceptions.propertyKeyCanNotBeNull();
```
Comment (velo) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientElement.java]: OrientDB can handle nulls, tinkerpop can not. 

Who has the final say?

May be we need a patch on gremlin to made this a feature

**Comment by velo** [comment]
Tests are almost 100%
````Tests run: 816, Failures: 21, Errors: 19, Skipped: 369````

Just take a look and if you agree with my changes, go head

**Comment by velo** [comment]
Latest numbers:
````Tests run: 818, Failures: 17, Errors: 19, Skipped: 369````
2 failures are true failures one transaction related and the other related to edge removal!

**Comment by velo** [comment]
last numbers:
````Tests run: 818, Failures: 2, Errors: 12, Skipped: 369````

now we have pretty much ALL tests on OrientGraphStructureStandardIT fixed!

the two failures are true bugs on this driver

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java (hunk 2 above)
Comment (mpollmeier) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java]: I find that rather odd. And I guess you do too, as you asked for my thoughts. 
I much prefer positive tests (A does X) over negative tests (A does not do X), as you can spend the rest of your life on the negative tests and only end up with messy code. And in this case, it's also iterating through the edges prematurely, just for that test. 
Can we selectively disable that test?

**Comment by mpollmeier** [comment]
awesome, great work! do you wanna fix the two bugs you are referring to and then we'll merge this?

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java (hunk 2 above)
Comment (velo) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java]: only by if we disable features.... but that would also turn off a huge set of tests...

well, I can try to be more selective them that.... HACKER MODE ON =D

**Comment by velo** [comment]
@mpollmeier I tried, but I failed miserably =/

one is transaction related... if you say you support transaction feature, it will turn it on (randomly), but that break tests since we try to change the schema... and that can only be done transactionless

The other, is the edge remove(), goes into an infinite loop

**Comment by mpollmeier** [comment]
hmm, I can have a look, but will have to wait until tomorrow. 
Just saw that you're based in Wellington - I'm in Auckland :) Your github profile still says Brazil...

**Comment by velo** [comment]
Fixed!

**Comment by mpollmeier** [comment]
did you amend the last commit then? sounds great!
however, I can't seem to run the tests locally, they always freeze. I run `mvn install` to execute them...

**Comment by velo** [comment]
no, I mean my github profile =D

**Comment by velo** [comment]
ok, fixed the issue we discussed!

feel free to merge if you agree with my changes.

**Comment by mpollmeier** [comment]
i still can't run them locally on my machine - what's your trick?

**Comment by velo** [comment]
Serious? Why not? Any error or what?

**Comment by mpollmeier** [comment]
it just randomly hangs at one test most of the time. I just tried it again though and the tests completed fine. output: `Tests run: 818, Failures: 2, Errors: 12, Skipped: 373`

So there's still a few issues?

**Comment by velo** [comment]
The errors are groovy tests related.... what I got 100% are the structure tests.  Anyway, the broken groovy tests need to be fixed on tinkerpop itself

**Pull request merged**